src/components/orgs/Coupon/Container.tsx

```diff
@@ -200,44 +205,50 @@ const CouponContainer: React.FC<CouponContainerProps> = ({
   };
 
   const handleActiveCoupon = async () => {
-    handleNextStep(STEPS.LOADING_COUPON);
+    if (session?.user) {
+      handleNextStep(STEPS.LOADING_COUPON);
 
-    activateCoupon()
-      .then((res) => {
-        if (res?.data?.coupon?.code) {
-          return setTimeout(() => {
-            setCouponCode(res?.data?.coupon?.code);
-          }, 2500);
-        }
+      activateCoupon()
+        .then((res) => {
+          if (res?.data?.coupon?.code) {
+            return setTimeout(() => {
+              setCouponCode(res?.data?.coupon?.code);
+            }, 2500);
+          }
 
-        setCurrentStep(STEPS.SHOWING_COUPON);
-        showToastify({
-          label: "Ocorreu um erro interno. Por favor, tente novamente.",
-          type: "error",
-        });
-      })
-      .catch((error) => {
-        setCurrentStep(STEPS.SHOWING_COUPON);
-        if (error?.response?.data?.code === 400) {
-          showToastify({
-            label:
-              "O Coupon que está tentando ativar não é mais válido, atualize a página.",
-            type: "error",
-          });
-        }
-        if (error?.response?.data?.code === 500) {
+          setCurrentStep(STEPS.SHOWING_COUPON);
           showToastify({
             label: "Ocorreu um erro interno. Por favor, tente novamente.",
             type: "error",
           });
-        }
-        if (error?.response?.data?.code === 404) {
-          showToastify({
-            label: "Nenhum dado encontrado. Por favor, tente novamente.",
-            type: "error",
-          });
-        }
+        })
+        .catch((error) => {
+          setCurrentStep(STEPS.SHOWING_COUPON);
+          if (error?.response?.data?.code === 400) {
+            showToastify({
+              label:
+                "O Coupon que está tentando ativar não é mais válido, atualize a página.",
+              type: "error",
+            });
+          }
+          if (error?.response?.data?.code === 500) {
+            showToastify({
+              label: "Ocorreu um erro interno. Por favor, tente novamente.",
+              type: "error",
+            });
+          }
+          if (error?.response?.data?.code === 404) {
+            showToastify({
+              label: "Nenhum dado encontrado. Por favor, tente novamente.",
+              type: "error",
+            });
+          }
+        });
+    } else {
+      signIn(undefined, {
+        callbackUrl: `/supplier-dashboard/${supplierId}/?coupon=${CouponId}`,
```

Comment:
```suggestion
        callbackUrl: `/supplier-dashboard/${supplierId}/?coupon=${couponId}`,
```

Please watch out case